### ISSUE TITLE
Preserve file type instead of copy

### ISSUE THREAD

**Issue by Xeite** [lra/mackup#1105]
I understand the process of backup from source below.

1. cp existed config file to synced folder
2. remove existed file.
3. link config file targeting to it in synced folder.

The problem is in the step 1, cp removes file type of original file. For example, if original file was symbolic linked file, copied file was not longer symbolic linked file. 

I think it would be better to preserve for file type since users has their own directory for config using symlink.

**Comment by huyz** [comment]
Yes, I was quite surprised how destructive an initial `mackup backup` was.  It destroyed my entire symlink history.  A warning would have been nice.
